internal/ethapi/api.go

```diff
@@ -1566,6 +1567,17 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 	signer := types.MakeSigner(s.b.ChainConfig(), bigblock)
 	from, _ := types.Sender(signer, tx)
 
+	// find l1 fee
+	message := types.NewMessage(from, tx.To(), tx.Nonce(), tx.Value(), tx.Gas(), tx.GasPrice(), tx.GasFeeCap(), tx.GasTipCap(), tx.Data(), tx.AccessList(), false)
+	state, _, err := s.b.StateAndHeaderByNumber(ctx, rpc.BlockNumber(blockNumber))
```

Comment:
It is not correct to use `state` from `blockNumber`, since the `l1BaseFee`, `scalar` and `overhead` could be changed in the block. Unless we have some extra assumption about the updating of `l1BaseFee`, `scalar` and `overhead`.

The most correct way is to add `l1Fee` in the receipt struct after transaction execution.

Reply:
FYI from discussion with Max

> `state, _, err := s.b.StateAndHeaderByNumber(ctx, rpc.BlockNumber(blockNumber))` will return the state after all the transactions in the block are executed.
> If there is a `l1BaseFee/scalar/overhead` updating tx (oracle updating tx) after your querying tx, the actual `l1Fee` should be different.
> You can reference the implementation of [optimism](https://github.com/ethereum-optimism/optimism/blob/develop/l2geth/internal/ethapi/api.go#L1486).